plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -837,13 +920,53 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
                        ("genesis", genesis_file.generic_string()));
 
             my->chain_config->genesis = fc::json::from_file( genesis_file ).as<genesis_state>();
+            const auto chain_id_from_gs = my->chain_config->genesis->compute_chain_id();
+            if ( !my->chain_id ) {
```

Comment:
Need to move checking and setting chain_id after all genesis parameters are handled.